src/XtensaOptimize.cpp

```diff
@@ -0,0 +1,2331 @@
+
+            std::vector<Expr> concat_args;
+            for (int ix = 0; ix < split_to; ix++) {
+                Expr r = Broadcast::make(value, native_lanes);
+                concat_args.push_back(std::move(r));
+            }
+            return Call::make(op->type,
+                              "halide_xtensa_concat_from_native",
+                              concat_args, Call::PureExtern);
+        }
+
+        return IRMutator::visit(op);
+    }
+
+    Expr visit(const Select *op) override {
+        int native_lanes = get_native_vector_lanes_num(op->type);
+        if (native_lanes > 0) {
+            const int total_lanes = op->type.lanes();
+            int split_to = op->type.lanes() / native_lanes;
+            Expr cond = mutate(op->condition);
+            Expr t = mutate(op->true_value);
+            Expr f = mutate(op->false_value);
+
+            std::vector<Expr> concat_args;
+            for (int ix = 0; ix < split_to; ix++) {
+                Expr sliced_cond = Call::make(cond.type().with_lanes(native_lanes),
+                                              "halide_xtensa_slice_to_native",
+                                              {cond, ix, native_lanes, total_lanes},
+                                              Call::PureExtern);
+                Expr sliced_t = Call::make(t.type().with_lanes(native_lanes),
+                                           "halide_xtensa_slice_to_native",
+                                           {t, ix, native_lanes, total_lanes},
+                                           Call::PureExtern);
+                Expr sliced_f = Call::make(f.type().with_lanes(native_lanes),
+                                           "halide_xtensa_slice_to_native",
+                                           {f, ix, native_lanes, total_lanes},
+                                           Call::PureExtern);
+                Expr r = Select::make(sliced_cond, sliced_t, sliced_f);
+                concat_args.push_back(std::move(r));
+            }
+            return Call::make(op->type,
+                              "halide_xtensa_concat_from_native",
+                              concat_args, Call::PureExtern);
+        }
+
+        int width_to_extend = get_width_to_extend(op->type);
+        if (width_to_extend > 0) {
+            const int lanes = op->type.lanes();
+
+            Expr cond = mutate(op->condition);
+            Expr t = mutate(op->true_value);
+            Expr f = mutate(op->false_value);
+
+            Expr padded_cond = pad(cond, lanes, width_to_extend);
+            Expr padded_t = pad(t, lanes, width_to_extend);
+            Expr padded_f = pad(f, lanes, width_to_extend);
+
+            Expr r = Select::make(padded_cond, padded_t, padded_f);
+
+            return slice(r, op->type, lanes);
+        }
+
+        return IRMutator::visit(op);
+    }
+
+    // NOTE(vksnk): not very clear if it's a good idea to slice loads/stores.
+    // Expr visit(const Load* op) override {
```

Comment:
This seems like a lot of commented-out code to check into main. Should probably be removed?